source/plugins/go/src/network_flow_logs.go

```diff
@@ -0,0 +1,372 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/fluent/fluent-bit-go/output"
+	"github.com/google/uuid"
+	"github.com/tinylib/msgp/msgp"
+)
+
+// Stream name for retina networkflow logs
+const RetinaNetworkFlowLogsStreamName = "RETINA_NETWORK_FLOW_LOGS"
+
+var (
+	// retina networkflow logs stream tag name
+	MdsdNetworkFlowLogsStreamTagName string
+	// flag to check whether network flow logs is enabled or not
+	IsNetworkFlowLogsEnabled bool
+)
+
+var (
+	// networkflow telemetry
+	NetworkFlowTelemetryMutex             = &sync.Mutex{}
+	NetworkFlowLogsMDSDClientCreateErrors float64
+	MdsdNetworkFlowClient                 net.Conn
+	NetworkFlowTagRefreshTracker          time.Time
+)
+
+// NetworkFlowMsgPackEntry represents the object corresponding to a single messagepack event in the messagepack stream
+type NetworkFlowMsgPackEntry struct {
+	Time   int64                  `msg:"time"`
+	Record map[string]interface{} `msg:"record"`
+}
+
+// PostNetworkFlowRecords sends data to the mdsd and amacoreagent
+func PostNetworkFlowRecords(tailPluginRecords []map[interface{}]interface{}) int {
+	if IsNetworkFlowLogsEnabled {
+		start := time.Now()
+		var elapsed time.Duration
+		var bts int
+		var er error
+
+		var dataMap map[string]interface{}
+		var networkFlowLogsMsgPackEntries []NetworkFlowMsgPackEntry
+		numNetworkLogRecords := 0
+
+		for _, record := range tailPluginRecords {
+			networkFlowLogRecordInterface, err := convertFluentBitRecord(record)
+			if err != nil {
+				Log("Error::networkflow:: failed to convert fluent-bit record: %v", err.Error())
+				continue
+			}
+			networkFlowLogRecord, ok := networkFlowLogRecordInterface.(map[string]interface{})
+			if !ok {
+				Log("Error::networkflow:: failed to convert networkflow record to map[string]interface{}: %v", networkFlowLogRecordInterface)
+				continue
+			}
+			dataMap = make(map[string]interface{})
+			if err := mapNetworkFlowLogsToDataMap(dataMap, networkFlowLogRecord); err != nil {
+				Log("Error::networkflow:: failed to map networkflow logs to data map: %v", err.Error())
+				continue
+			}
+			var networkFlowLogsMsgPackEntry NetworkFlowMsgPackEntry
+			networkFlowLogsMsgPackEntry = NetworkFlowMsgPackEntry{
+				Time:   time.Now().Unix(),
+				Record: dataMap,
+			}
+			networkFlowLogsMsgPackEntries = append(networkFlowLogsMsgPackEntries, networkFlowLogsMsgPackEntry)
+		}
+
+		if len(networkFlowLogsMsgPackEntries) > 0 {
+			if IsAADMSIAuthMode == true {
```

Comment:
since this feature only works in MSI mode, you should IsAADMSIAuthMode  club with IsNetworkFlowLogsEnabled .  

if IsAADMSIAuthMode   && IsNetworkFlowLogsEnabled {

Reply:
moved